content/articles/visualizing-repositories-using-pygal/index.md

```diff
@@ -0,0 +1,227 @@
+    print('stars-:', repos_dict['stargazers_count'])#print the number of stars the project has
+    print('repo-:', repos_dict['html_url'])  # print the link to the project's repository
+    print('Description-:', repos_dict['description']) # print the description of the project
+   
+```
+
+We print the owner of each project, its name, the number of stars it has, the project's description, and its GitHub URL inside the loop. Click [here](https://github.com/duncandegwa/visualizing-repositories-using-pygal/blob/main/visualizing.md) to see the output of the above code.
+
+Let's construct a visual representation of the current popularity of Python works on GitHub since we have some relevant data. We will create a bar chart: the tally of stars the project has earned will be represented by the height of each bar.
+To get this done, you can run the code below:
+
+```python
+import requests
+import pygal
+from pygal.style import LightColorizedStyle as LCSe, LightenStyle as LSe
+#  Start an API request.
+url = 'https://api.github.com/search/repositories?q=language:python&sort=star'
+request = requests.get(url)
+print("Status  code:", request.status_code)
+# In a variable, save the API response.
+response_dictionary = request.json()
+print("Total-repos:", response_dictionary['total_count'])
+# Investigate the repository's contents.
+repos_dicts = response_dictionary['items']
+names, stars = [], []
+for repos_dict in repos_dicts:
+    names.append(repos_dict['name'])
+    stars.append(repos_dict['stargazers_count'])
+# Construct a visual picture
+our_style = LSe('#333366', base_style=LCSe)
+chart = pygal.Bar(style=our_style, x_label_rotation=45, show_legend=False)
+chart.title = 'Python Works with the Most Stars on GitHub'
+chart.x_labels = names
+chart.add('', stars)
+chart.render_to_file('py_repos.svg')
+```
+
+To begin, we `import pygal` as well as the `Pygal styles` we'll require for the chart. We'll keep printing the status of the API request return and the total amount of repositories detected to see if the API call was successful. 
+
+First, we generate 2 blank lists to house the data that will be used in the chart. We’ll use the name, and the number of stars of each project to label and determine the height of the bars respectively. The name of each work, as well as the amount of stars it has, are added to these lists inside the loop.
+
+Next, we create a style based on a blue colored tint using the `LightenStyle` class `(alias LSe)`.In order to use the `LightColorizedStyle class`(`alias LCSe`), we additionally give the base style parameter. After that, we create a simple chart with Bar() and provide it our_style.We further give two extra style arguments: we rotate the labels 45 degrees along the x-axis (`x label rotation=45 `) and conceal the legend since we're just showing one series on the graph (`show legend=False`). The `x labels` property is assigned to the list names, and the graph is given a title. 
```

Comment:
This section is a bit difficult to read though, maybe for beginners, you can use some  connectors to break complexity.